HiP-DataStore/Controllers/ExhibitsController.cs

```diff
@@ -193,6 +195,73 @@ public IActionResult GetById(int id, DateTimeOffset? timestamp = null)
             return NoContent();
         }
 
+        [HttpGet("rating/{id}")]
+        [ProducesResponseType(typeof(RatingResult), 200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public IActionResult GetRating(int id)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            var exhibit = _db.Database.GetCollection<Exhibit>(ResourceType.Exhibit.Name)
+                                      .AsQueryable()
+                                      .FirstOrDefault(x => x.Id == id);
+
+            if (exhibit == null || _entityIndex.Status(ResourceType.Exhibit, id) != ContentStatus.Published)
+                return NotFound(ErrorMessages.ExhibitNotFoundOrNotPublished(id));
+
+            var result = new RatingResult()
+            {
+                Id = id,
+                Average = _ratingIndex.Average(ResourceType.Exhibit, id),
+                Count = _ratingIndex.Count(ResourceType.Exhibit, id)
+            };
+
+            return Ok(result);
+        }
+
+        [HttpPost("rating/{id}")]
+        [ProducesResponseType(typeof(int),201)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> PostRating(int id,RatingArgs args)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            var exhibit = _db.Database.GetCollection<Exhibit>(ResourceType.Exhibit.Name)
+                                      .AsQueryable()
+                                      .FirstOrDefault(x => x.Id == id);
+
+            if (exhibit == null || _entityIndex.Status(ResourceType.Exhibit,id) != ContentStatus.Published)
+                return NotFound(ErrorMessages.ExhibitNotFoundOrNotPublished(id));
+
+            var oldValue = _db.Database
+                              .GetCollection<Rating>(ResourceType.Rating.Name)
+                              .AsQueryable()
+                              .Where(x => x.UserId == args.UserId.GetValueOrDefault() && x.RatedType.Name == ResourceType.Exhibit.Name && x.EntityId == id)
```

Comment:
You can directly check `x.RatedType == ResourceType.Exhibit`, leaving out `.Name` on both sides